### PR TITLE
Initial commit of estimator component

### PR DESCRIPTION
**Pull request by rgeyer**
#### PR Description
Adds a new component for estimating/reporting the "amount" of a given billable unit is being received.

For metrics, the number of active series.
For logs and traces, the byte throughput/counter.

The component will also expose a prometheus exporter interface, and a list of targets which can be scraped and remote written for analysis.

#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: component/local/estimator/estimator.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	// Could consume less memory by combining those two values into a single unique hash, but I'm not sure it's worth the additional effort ¯\_(ツ)_/¯
+	// Also, we may end up storing *actual* values for these, should we wish to report on them individually. I.E. There are x series with y label(s)
+	activeSeries map[storage.SeriesRef]map[uint64]struct{}
+}
+
+type Arguments struct {
+}
+
+type Exports struct {
+	metricsReceiver *prometheus.Interceptor `river:metrics_reciever,attr`
```
Comment (cyriltovena) [inline component/local/estimator/estimator.go:37]:
```suggestion
	metricsReceiver *prometheus.Interceptor `river:metrics_receiver,attr`
```

Comment (rgeyer) [reply component/local/estimator/estimator.go:37]: Spelling is hard 😂 

This is getting refactored anyhow, but, I have fixed the spelling :)

**Comment by tpaschalis** [comment]
Hey there @rgeyer 👋 

Do you think we should make a push for including these new components (adding documentations, tidying code up etc), or was it more of a proof of concept?

I'm going through a round of tidying up our PR queue and was thinking of closing this otherwise.

**Comment by rgeyer** [comment]
> Hey there @rgeyer 👋
> 
> Do you think we should make a push for including these new components (adding documentations, tidying code up etc), or was it more of a proof of concept?
> 
> I'm going through a round of tidying up our PR queue and was thinking of closing this otherwise.

Only the metrics portion was "correct". The logging functionality is incorrect, and I'm not sure I'd be able to resolve it.

That said, the work that @captncraig did for the most recent hackathon (x-ray), I think this becomes superflous. Also, x-ray seems more complete and useful.

I think we can close this.

**Pull request closed without merging**